src/sayt/products.ts

```diff
@@ -7,6 +7,8 @@ export const SAYT_PRODUCTS_REQUEST = 'gbe::sayt_products_request';
 export interface SaytProductsRequestPayload extends WithQuery, WithGroup {
   /** The extra configuration options to customize the request. */
   config?: Partial<Request>;
+  /** The origin of the products request action. */
+  origin: string;
```

Comment:
Do we want this to be any arbitrary string, or should we restrict it to a known set of values using an enum or a union of string literal types? Maybe even `keyof SendableOrigin` if we want that kind of coupling?

Reply:
I had thought about making it more specific, but I didn't want to manually list a bunch of options and have to update them later. `keyof SendableOrigin` is a good idea.